posts/TEMPLATE/index.md

```diff
@@ -1,9 +1,49 @@
 ---
 date: 2030-12-31 # Auto updates on publish.
-title: Hello World!
-author: sascha.lampalzer@cybertec.at
-tags: ["list-tags-here", "🙂"] # max. 10 tags; lowercase; dash-separated
-description: "This is an example" # max. 300 chars.
+title: Using podman in bridged network mode
+author: julian.markwort@cybertec.at
+tags: ["podman-network-bridge-bridged-ip-address-docker-slirp4netns", "🙂"] # max. 10 tags; lowercase; dash-separated
+description: "Opening ports in podman is simple: add a -p or --publish to your podman run command and specify which ports to direct where and you're off. But what if you want your container to be reachable from your host by ip address?" # max. 300 chars.
 ---
 
-This is just an example post, don't worry about it. We use this internally as a template for awesome, future posts.
+Podman is really great for those of us who don't want the docker daemon running in the background all the time.
+
+It is mostly compatible with Dockerfiles and Syntax (as far as I've read online and noticed, while poking at both of them), but some things are handled differently due to the nature of podman's daemonless architecture.
+
+I was trying to create a couple of docker containers with IP addresses so I could test some ansible scripts.
+I have to make sure the scripts work for different distributions and different versions of those distributions and managing virtual machines (or even clusters of those) in virtualbox can become a bit frustrating.
+
+So for ansible, I usually need hostnames or IP adresses to define against which hosts my playbooks are run.
+But whenever I was running containers, they didn't have IP addresses:
+```
+[julian@localhost ansible]$ podman run -dit --name centos1 centos:7
+faf5abe92901c4757982bbcb39f0a89800e7378358fff88908bea09161922282
+[julian@localhost ansible]$ podman inspect centos1 | grep -i ipaddress
+            "SecondaryIPAddresses": null,
+            "IPAddress": "",
+```
+
+Of course, I could open up ports by publishing them with the -p flag, but that won't help me to use ansible.
+
+#### Google to the rescue??
+
+Googling turned up no helpful advice, I saw that the `--net host` option is dangerous and should be used with caution, but no help on bridged networking, as I know it from virtualbox.
+
+After reading the [docs](https://github.com/containers/libpod/blob/master/docs/podman-run.1.md) on `podman run`, I figured it out: if you're running podman without root, it can only use the network mode `slirp4netns`, which will create an isolated network stack so that you can connect to the internet from the inside of your container and bind certain ports of your container to the user-bindable ports on you host, but nothing more.
```

Comment:
```suggestion
After reading the [docs](https://github.com/containers/libpod/blob/master/docs/podman-run.1.md) on `podman run`, I figured it out: if you're running Podman without root, it can only use the network mode `slirp4netns`, which will create an isolated network stack so that you can connect to the internet from the inside of your container and bind certain ports of your container to the user-bindable ports on you host, but nothing more.
```

Reply:
> 

```suggestion
After reading the [docs](https://github.com/containers/libpod/blob/master/docs/podman-run.1.md) on `podman run`, I figured it out: if you're running Podman without root, it can only use the network mode `slirp4netns`, which will create an isolated network stack so that you can connect to the internet from the inside of your container and bind certain ports of your container to the user-bindable ports on your host, but nothing more.
```